### PR TITLE
sidemenu added for issue #14

### LINKED ISSUE #14: Storybook: Side Structural Menu
## User story

**Who:** **As a** user
**What:** **I want to** be able to use the options on the side menu 
**Why:** **so that I can** I can choose between **Home**, **Next Steps**, **Elevator Pitch**

## Acceptance criteria

Create a storybook component **sidebar utility menu** like the one we can see highlighted in pink where we can see which screen/page of the menu we are at by checking which one is highlighted with vertical white bar.
 
Screenshot 2021-04-13 at 07 50 27

Screenshot 2021-04-13 at 07 58 35

Screenshot 2021-04-13 at 08 30 42

Check the wireframes here: https://www.figma.com/file/9SJRaxPSuPI3lSl8ULNyJ8/GuideIT-For-Developers?node-id=0%3A1

## Implementation details

Use the font [Chakra Petch](https://fonts.google.com/specimen/Chakra+Petch?preview.text=GuideIT&preview.text_type=custom).  The text color must be #FFFFFF and the background color should be #298F8A.

Use the images below and add the files to the project on this folder /src/client/assets/images/

Elevator_btn
NextSteps_btn
Home_btn

### PR DESCRIPTION
**Pull request by ccshah298**
# Description
side structural menu added

Fixes #14 

# How to test?

Please provide a short summary how your changes can be tested?
npm run storybook

# Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/sideMenubar/sideMenu.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          <li className="side_menu_spacebetween">
+            <a href="#dummylink">Next steps</a>
+          </li>
+          <li className="side_menu_whiteborder">
+            <img src={elevatorImg} alt="elevatorImg" />
+          </li>
+          <li className="side_menu_spacebetween">
+            <a href="#dummylink">Elevetor pitch</a>
+          </li>
+        </ul>
```
Comment (cph-kiwi) [inline src/client/components/sideMenubar/sideMenu.js:30]: The little indicator bar is not working as intended. Currently it points to the icon that the mouse is hovering over. The indicator should match the page it is displaying. To do this, you need to add conditional logic into your .js file. In simple words - if "Page Title" is equal to what the icon represents, then indicator bar goes next to icon.

File: src/client/components/sideMenubar/sideMenu.styles.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+li {
+  background-color: #298f8a;
+  text-align: center;
+  padding-top: 3px;
+  border-right: solid 5px transparent;
+  line-height: 0.8;
+}
+
+.side_menu_whiteborder:hover {
```
Comment (cph-kiwi) [inline src/client/components/sideMenubar/sideMenu.styles.css:15]: This is the part that is controlling the indicator bar at the moment, so you need to replace this with logic in the .js file.

**Review by cph-kiwi — CHANGES_REQUESTED** [review]

File: src/client/components/sideMenubar/sideMenu.styles.css (hunk 2 above)
Comment (ccshah298) [reply src/client/components/sideMenubar/sideMenu.styles.css:15]: Hi Beth Thanks for reviewing my PR.. I will change that part.. Thanks again :)

File: src/client/components/sideMenubar/sideMenu.styles.css [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
   border-right-color: white;
 }
 
 .side_menu_ul {
+  width: 100%;
   list-style: none;
-  padding-left: 15px;
-  padding-right: 15px;
+  margin: 10px auto;
 }
```
Comment (TITHI1244) [inline src/client/components/sideMenubar/sideMenu.styles.css:24]: I think, a little more padding could be added to avoid the texts touching border :)

**Review by TITHI1244 — COMMENTED** [review]
The top-most list item has some extra padding of 15px. And that's why whenever 'home' is hovered it shows a longer border than other 2. To avoid it you can add 'side_menu_whiteborder' class to the image :)
You can also add a control (type select) that controls which screen/page we are at now. Then according to that, the vertical white border comes @ccshah298  :) Good job.

File: src/client/components/sideMenubar/sideMenu.styles.css (hunk 3 above)
Comment (ccshah298) [reply src/client/components/sideMenubar/sideMenu.styles.css:24]: Thanks Tithi for reviewing and giving suggestions :)

**Review by cph-kiwi — APPROVED** [review]
I see now that @ccshah298 has aimed for the appearance of the component. Functionality can be added later.

File: src/client/components/sideMenubar/sideMenu.styles.css (hunk 3 above)
Comment (ccshah298) [reply src/client/components/sideMenubar/sideMenu.styles.css:35]: hi @TITHI1244 and @cph-kiwi I tried to make changes as per your comment ... Thamks :)

File: src/client/components/sideMenubar/sideMenu.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <ul className="side_menu_ul">
+          <li
+            className="side_menu_spacebetween side_menu_whiteborder"
+            data-highlighted={highLigtItem === 0}
+          >
+            <a href={highLigtItem}>
+              <img src={homeImg} alt="homeImg" />
+            </a>
+          </li>
+          <li className="side_menu_spacebetween">Home</li>
```
[the remark this answers is not in the document]
Comment (wwang107) [reply src/client/components/sideMenubar/sideMenu.js:21]: Also give it a nav label class and configure the spacing between the label and icon there. Currently, the label has the same margin top and bottom. However, it should be closer to the icon where it belongs to and further from the icon in the bottom.

[the remark this answers is not in the document]
Comment (ccshah298) [reply src/client/components/sideMenubar/sideMenu.js]: Thanks for reviewing my PR.. I added changes as per your comment.. Thanks again. :)

File: src/client/components/sideMenubar/sideMenu.styles.css [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  width: 65px;
+  background-color: #298f8a;
+  margin-left: 50px;
+  padding: 0;
+  color: #fff;
+  font-size: x-small;
+  font-family: 'Chakra Petch', monospace;
+}
+
+li {
```
Comment (bytewiz) [inline src/client/components/sideMenubar/sideMenu.styles.css:17]: Just be careful with these "global" tags

**Review by bytewiz — APPROVED** [review]

**Pull request merged**